pkg/action/package.go

```diff
@@ -104,6 +109,14 @@ func (p *Package) Run(path string, vals map[string]interface{}) (string, error)
 	return name, err
 }
 
+// validateName verifies if the chart name has any illegal characters or not
+func validateName(name string) error {
```

Comment:
This function doesn't validate the same as we do elsewhere. See https://github.com/helm/helm/blob/26ce96464f73ca72d0855cf49af31bac47f38bb7/pkg/chartutil/create.go#L679-L687

I think we need to be consistent throughout Helm. Can you please update this?

I also wonder if this should become a public function (with tests).

Reply:
> I also wonder if this should become a public function (with tests).

I agree.

But, the chart name regex here does not match the requirements mentioned in the doc; 
https://github.com/helm/helm/blob/26ce96464f73ca72d0855cf49af31bac47f38bb7/pkg/chartutil/create.go#L39
https://helm.sh/docs/chart_best_practices/conventions/#chart-names

Reply:
@mattfarina Not sure whether to change regex to what's mentioned in the doc or not.